src/lib/search-and-replace.js

```diff
@@ -0,0 +1,124 @@
+/**
+ * @flow
+ * @format
+ */
+
+/**
+ * External dependencies
+ */
+import fs from 'fs';
+import os from 'os';
+import path from 'path';
+import util from 'util';
+import chalk from 'chalk';
+import debugLib from 'debug';
+import suffix from 'suffix';
+import { replace } from '@automattic/vip-search-replace';
+
+/**
+ * Internal dependencies
+ */
+import { trackEvent } from 'lib/tracker';
+import { confirm } from 'lib/cli/prompt';
+import { getFileSize } from 'lib/client-file-uploader';
+
+const debug = debugLib( '@automattic/vip:lib:search-and-replace' );
+
+const flatten = arr => {
+	return arr.reduce( function( flat, toFlatten ) {
+		return flat.concat( Array.isArray( toFlatten ) ? flatten( toFlatten ) : toFlatten );
+	}, [] );
+};
+
+const inPlaceReplacement = async filename => {
+	await confirm(
+		[],
+		'Are you sure you want to run search and replace on your input file? This operation is not reversible.'
+	);
+
+	const tmpFilePath = path.join( os.tmpdir(), ( +new Date() ).toString( 36 ) );
+
+	const copyFile = util.promisify( fs.copyFile );
+	await copyFile( filename, tmpFilePath );
+
+	return {
+		inputFile: tmpFilePath,
+		outputFile: filename,
+	};
+};
+
+const outputFileReplacement = filename => {
```

Comment:
Similarly, the fn name `outputFileReplacement` doesn't make it clear to me that this is returning a mapping of file names.